packages/devtools_app/lib/src/screens/memory/panes/allocation_tracing/allocation_profile_tracing_view_controller.dart

```diff
@@ -83,6 +86,18 @@ class AllocationProfileTracingViewController extends DisposableController
   final _tracedClasses = <String, TracedClass>{};
   final _tracedClassesProfiles = <String, CpuProfileData>{};
 
+  void updateClassFilter(String value) {
+    if (value.isEmpty && _currentFilter.isEmpty) return;
+    _currentFilter = value;
+    final updatedFilter = _unfilteredClassList
```

Comment:
we could be smarter here and only look through the filtered data when the previous value `_currentFilter` is a substring of the new value `value`